components/log4j/org.wso2.carbon.analytics.shared.data.agents.log4j/src/main/java/org/wso2/carbon/analytics/shared/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -171,11 +143,49 @@ public void activateOptions() {
         }
     }
 
+    private void publisherInitializer() {
+        Properties passwordProperty = new Properties();
+        passwordProperty.put("log4j.appender.DAS_AGENT.password", password);
+        SecretResolver secretResolver = SecretResolverFactory.create(passwordProperty);
+        String secretAlias = "log4j.appender.DAS_AGENT.password";
+        //Checking the log4j appender DAS credentials.
+        if (secretResolver != null && secretResolver.isInitialized()) {
+            if (secretResolver.isTokenProtected(secretAlias)) {
+                password = secretResolver.resolve(secretAlias);
+            } else {
+                password = (String) passwordProperty.get(secretAlias);
+            }
+        }
+        try {
+            dataPublisher = new DataPublisher("Thrift", url, authURLs, userName, password);
+        } catch (DataEndpointAgentConfigurationException e) {
+            log.error(
+                    "Invalid urls passed for receiver and auth, and hence expected to fail " + e
+                            .getMessage(), e);
+        } catch (DataEndpointException e) {
+            log.error(
+                    "Error while trying to publish events to data receiver " + e
+                            .getMessage(), e);
+        } catch (DataEndpointConfigurationException e) {
+            log.error(
+                    "Invalid urls passed for receiver and auth, and hence expected to fail " + e
+                            .getMessage(), e);
+        } catch (DataEndpointAuthenticationException e) {
+            log.error(
+                    "Error while trying to login to data receiver : " + e
+                            .getMessage(), e);
+        } catch (TransportException e) {
+            log.error(
+                    "Thrift transport exception occurred " + e
+                            .getMessage(), e);
+        }
```

Comment:
Hi Tishan,

This formatting is fixed by the commit in https://github.com/wso2/shared-analytics/commit/38adb47c33b1541fc238c86475269ab5fa507c6d

Thanks,
SajithD